app/src/main/java/com/infomaniak/drive/ui/home/RootFileTreeCategory.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Infomaniak kDrive - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.drive.ui.home
+
+enum class RootFileTreeCategory {
+    // Order is IMPORTANT, not for UI, but for storage consistency.
+    // Don't remove entries, only rename and/or deprecate them.
+    // New entries must be added at the end.
+    CommonFolders,
+    PersonalFolder,
+    Favorites,
+    RecentChanges,
+    SharedWithMe,
+    MyShares,
+    Offline,
+    Trash;
```

Comment:
YOU_MEAN_LIKE_THIS? 😄 

From [Kotlin's official coding conventions](https://kotlinlang.org/docs/coding-conventions.html#property-names) (emphasis mine):

> For enum constants, it's OK to use either all uppercase, underscore-separated ([screaming snake case](https://en.wikipedia.org/wiki/Snake_case)) names (enum class Color { RED, GREEN }) **or upper camel case names**, depending on the usage.

I prefer "NonScreaming" for 3 reasons:
1. It is not putting unbalanced emphasis, helping reading
2. It is more common in reference Kotlin projects
3. It It aligns with the case used in other kinds of symbols that are used for use cases similar to enum entries: objects in sealed classes & interfaces, constants…

Any chance we can start adopting UpperCamelCase instead of UPPER_SNAKE_CASE for new code, starting with this PR? 🙏

Reply:
We'll talk about it in the team meeting to see what we want to do from now on